### PR TITLE
PyPy: Fix UB when passing inputs to C

### PR DESCRIPTION
**Pull request by ax3l**
Fix the macOS PyPy 3.7 error seen in #37 #36

Patch from https://github.com/ECP-WarpX/WarpX/pull/2726

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ax3l** [comment]
Was after debug still:
```
amrex_init argv= ['warpx', 'warpx.verbose = 1', 'warpx.use_filter = 1', 'warpx.serialize_ics = 1', 'warpx.do_dynamic_scheduling = 0', 'warpx.do_moving_window = 1', 'warpx.moving_window_dir = z', 'warpx.moving_window_v = 1.0', 'warpx.cfl = 1.0', 'warpx.do_dive_cleaning = 0', 'amr.n_cell = 32 32 256', 'amr.max_grid_size = 64', 'amr.blocking_factor = 32', 'amr.max_level = 0', 'geometry.dims = 3', 'geometry.prob_lo = -3e-05 -3e-05 -5.6e-05', 'geometry.prob_hi = 3e-05 3e-05 1.2e-05', 'boundary.field_lo = periodic periodic pec', 'boundary.field_hi = periodic periodic pec', 'boundary.particle_lo = periodic periodic absorbing', 'boundary.particle_hi = periodic periodic absorbing', 'algo.particle_shape = 3', 'algo.maxwell_solver = Yee', 'particles.species_names = electrons', 'electrons.mass = m_e', 'electrons.charge = -q_e', 'electrons.injection_style = nuniformpercell', 'electrons.initialize_self_fields = 0', 'electrons.num_particles_per_cell_each_dim = 1 1 1', 'electrons.xmin = -2e-05', 'electrons.xmax = 2e-05', 'electrons.ymin = -2e-05', 'electrons.ymax = 2e-05', 'electrons.zmin = 0', 'electrons.profile = constant', 'electrons.density = 2e+23', 'electrons.momentum_distribution_type = constant', 'electrons.ux = 0.0', 'electrons.uy = 0.0', 'electrons.uz = 0.0', 'electrons.do_continuous_injection = 1', 'lasers.names = laser1', 'laser1.profile = Gaussian', 'laser1.wavelength = 8e-07', 'laser1.e_max = 16000000000000.0', 'laser1.polarization = 0 1 0', 'laser1.profile_waist = 5e-06', 'laser1.profile_duration = 1.5e-14', 'laser1.do_continuous_injection = 0', 'laser1.position = 0.0 0.0 9e-06', 'laser1.direction = 0 0 1', 'laser1.profile_focal_distance = 0.0001', 'laser1.profile_t_peak = 3e-14', 'diagnostics.diags_names = diag1', 'diag1.diag_type = Full', 'diag1.format = plotfile', 'diag1.intervals = 100', 'diag1.fields_to_plot = Bx By Bz Ex Ey Ez jx jy jz rho', 'diag1.file_prefix = ./Python_LaserAcceleration_plt']
amrex_init len(argv)= 59
len(argvC)= 59
argc= 59
argvC= <_ctypes.array.LP_c_char_Array_59 object at 0x00007fb2b1f89dc0>
amrex::Error::0::Couldn't open file: ��ή� !!!
SIGABRT
See Backtrace.0.0 file for details
```

**Comment by ax3l** [comment]
Now with `ascii` encoding enforced, also `linux_64_python3.7.____73_pypy` fails:
```
+ /home/conda/feedstock_root/build_artifacts/warpx_1641919919183/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_/bin/python Examples/Physics_applications/laser_acceleration/PICMI_inputs_3d.py
amrex_init argv= ['warpx', 'warpx.verbose = 1', 'warpx.use_filter = 1', 'warpx.serialize_ics = 1', 'warpx.do_dynamic_scheduling = 0', 'warpx.do_moving_window = 1', 'warpx.moving_window_dir = z', 'warpx.moving_window_v = 1.0', 'warpx.cfl = 1.0', 'warpx.do_dive_cleaning = 0', 'amr.n_cell = 32 32 256', 'amr.max_grid_size = 64', 'amr.blocking_factor = 32', 'amr.max_level = 0', 'geometry.dims = 3', 'geometry.prob_lo = -3e-05 -3e-05 -5.6e-05', 'geometry.prob_hi = 3e-05 3e-05 1.2e-05', 'boundary.field_lo = periodic periodic pec', 'boundary.field_hi = periodic periodic pec', 'boundary.particle_lo = periodic periodic absorbing', 'boundary.particle_hi = periodic periodic absorbing', 'algo.particle_shape = 3', 'algo.maxwell_solver = Yee', 'particles.species_names = electrons', 'electrons.mass = m_e', 'electrons.charge = -q_e', 'electrons.injection_style = nuniformpercell', 'electrons.initialize_self_fields = 0', 'electrons.num_particles_per_cell_each_dim = 1 1 1', 'electrons.xmin = -2e-05', 'electrons.xmax = 2e-05', 'electrons.ymin = -2e-05', 'electrons.ymax = 2e-05', 'electrons.zmin = 0', 'electrons.profile = constant', 'electrons.density = 2e+23', 'electrons.momentum_distribution_type = constant', 'electrons.ux = 0.0', 'electrons.uy = 0.0', 'electrons.uz = 0.0', 'electrons.do_continuous_injection = 1', 'lasers.names = laser1', 'laser1.profile = Gaussian', 'laser1.wavelength = 8e-07', 'laser1.e_max = 16000000000000.0', 'laser1.polarization = 0 1 0', 'laser1.profile_waist = 5e-06', 'laser1.profile_duration = 1.5e-14', 'laser1.do_continuous_injection = 0', 'laser1.position = 0.0 0.0 9e-06', 'laser1.direction = 0 0 1', 'laser1.profile_focal_distance = 0.0001', 'laser1.profile_t_peak = 3e-14', 'diagnostics.diags_names = diag1', 'diag1.diag_type = Full', 'diag1.format = plotfile', 'diag1.intervals = 100', 'diag1.fields_to_plot = Bx By Bz Ex Ey Ez jx jy jz rho', 'diag1.file_prefix = ./Python_LaserAcceleration_plt']
amrex_init len(argv)= 59
len(argvC)= 59
argc= 59
argvC= <_ctypes.array.LP_c_char_Array_59 object at 0x000056390b21bbb0>
amrex::Error::0::Couldn't open file: m !!!
```

File: recipe/build.sh [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
 # find out toolchain C++ standard
-CXX_STANDARD=14
+CXX_STANDARD=17
 if [[ ${CXXFLAGS} == *"-std=c++14"* ]]; then
-    echo "14"
-    CXX_STANDARD=14
+    echo "14 -> 17"
+    export CXXFLAGS="${CXXFLAGS} -std=c++17"
+    if [[ ! -z ${MACOSX_DEPLOYMENT_TARGET} ]]; then
+        export MACOSX_DEPLOYMENT_TARGET=10.13
```
Comment (isuruf) [inline recipe/build.sh:9]: Why this?

Comment (ax3l) [reply recipe/build.sh:9]: `-std=c++17`: to append behind the `-std=c++14` flag. CMake will not set extra flags if the [compiler default](https://gist.github.com/ax3l/53db9fa8a4f4c21ecc5c4100c0d93c94) is already C++17 and I request it.

`MACOSX_DEPLOYMENT_TARGET`: I thought to have seen that the default set here is smaller in the env, but I need `10.13`/`10.14` for C++17 support in the STDlib.

Comment (isuruf) [reply recipe/build.sh:9]: You don't need 10.13/10.14. can you revert that?

Comment (ax3l) [reply recipe/build.sh:9]: Sure!

**Comment by ax3l** [comment]
Dunno what the Windows issue is yet though with loading the lib location.

**Comment by isuruf** [comment]
https://github.com/conda-forge/warpx-feedstock/blob/master/recipe/build.sh#L46 should be changed to `bin`. DLLs should go in `bin`. Not sure why it works in 3.7 and 3.9.

**Comment by ax3l** [comment]
Ah right, it's Windows and `.dll`s need to live in `PATH` :see_no_evil:

**Comment by ax3l** [comment]
Hm, nope does not help: #39

I would be surprised as well, because we actually copy the `.dll` into our python package path...

**Pull request merged**